src/hooks/useLanguage.ts

```diff
@@ -1,14 +1,100 @@
 import { useCallback, useEffect, useState } from 'preact/hooks';
 import {
   ConsentManagerLanguageKey,
+  LanguageKey,
   TranslatedMessages,
   Translations,
 } from '@transcend-io/internationalization';
 import { settings } from '../settings';
 import { substituteHtml } from '../utils/substitute-html';
+import { invertSafe } from '@transcend-io/type-utils';
 
 export const loadedTranslations: Translations = Object.create(null);
 
+/**
+ * Mapping of browser locale to AWS base translation key
+ *
+ * TODO: https://transcend.height.app/T-39777
+ * This is here for a quick fix of our CM UI translation logic. Copied from the monorepo's extract_intl.ts
+ * This should be removed ASAP (it should be in the intl repo and imported, not in here or extract_intl.ts)
+ */
+export const TRANSLATE_LOCALE = {
+  [LanguageKey.EsEs]: 'es',
```

Comment:
Were our JSON filenames in the AWS language key format? Trying to understand where this format comes into play

Reply:
unfortunately no, its super confusing. there are aws language keys, there are transcend language keys, there are transcend -consent ui- language keys, then there are browser language keys. all subtly distinct, but we dont always treat them as such

Reply:
Where does the AWS language key format come into play, if it isn't being used for the JSON files?

Reply:
Oh I see now. ConsentManagerLanguageKey was mapped to LanguageKey